sdk/core/Azure.Core/src/Shared/TaskExtensions.cs

```diff
@@ -11,38 +12,69 @@ internal static class TaskExtensions
     {
         public static T EnsureCompleted<T>(this Task<T> task)
         {
-            Debug.Assert(task.IsCompleted);
+#if DEBUG
+            VerifyTaskCompleted(task.IsCompleted);
+#endif
+#pragma warning disable AZC0102
             return task.GetAwaiter().GetResult();
+#pragma warning restore AZC0102
         }
 
         public static void EnsureCompleted(this Task task)
         {
-            Debug.Assert(task.IsCompleted);
+#if DEBUG
+            VerifyTaskCompleted(task.IsCompleted);
+#endif
+#pragma warning disable AZC0102
             task.GetAwaiter().GetResult();
+#pragma warning restore AZC0102
         }
 
         public static T EnsureCompleted<T>(this ValueTask<T> task)
         {
-            Debug.Assert(task.IsCompleted);
+#if DEBUG
+            VerifyTaskCompleted(task.IsCompleted);
+#endif
+#pragma warning disable AZC0102
             return task.GetAwaiter().GetResult();
+#pragma warning restore AZC0102
         }
 
         public static void EnsureCompleted(this ValueTask task)
         {
-            Debug.Assert(task.IsCompleted);
+#if DEBUG
+            VerifyTaskCompleted(task.IsCompleted);
+#endif
+#pragma warning disable AZC0102
             task.GetAwaiter().GetResult();
+#pragma warning restore AZC0102
         }
 
         public static ConfiguredValueTaskAwaitable<T> EnsureCompleted<T>(this ConfiguredValueTaskAwaitable<T> task, bool async)
         {
+#pragma warning disable AZC0109
             Debug.Assert(async || task.GetAwaiter().IsCompleted);
+#pragma warning restore AZC0109
             return task;
         }
 
         public static ConfiguredValueTaskAwaitable EnsureCompleted(this ConfiguredValueTaskAwaitable task, bool async)
         {
+#pragma warning disable AZC0109
             Debug.Assert(async || task.GetAwaiter().IsCompleted);
```

Comment:
Use VerifyTaskCompleted everywhere.

Reply:
Will we need these methods after analyzer is updated?

Reply:
Ok, let's see how that looks.